pkg/codegen/schema/schema.go

```diff
@@ -1272,6 +1324,8 @@ func bindResource(token string, spec ResourceSpec, types *types) (*Resource, err
 		return nil, errors.Wrap(err, "failed to bind properties")
 	}
 
+	// TODO: Do we need to extend StateInputs to accept either an ObjectType or ResourceType?
+	// 		 This might require a new interface type.
```

Comment:
Nope, `StateInputs` should just be an `ObjectType`: it's describing the shape of the inputs to the resource's `.get` function.